internal/service/users/user.go

```diff
@@ -97,3 +97,14 @@ func (s *Service) findUser(user *types.User) (*types.User, error) {
 	}
 	return user, types.NewErrServerError(result.Error)
 }
+
+func (s *Service) CreateUserSponsorship(userId uuid.UUID, sponsorId uuid.UUID) (types.UserSponsorship, error) {
+
+	result := s.db.Create(&types.UserSponsorship{
+		UserID:    userId,
+		SponsorID: sponsorId,
+	})
```

Comment:
this should be idempotent – i.e. FirstOrCreate and assigning the CreatedAt time. we don't want multiple records with the same `UserID` and `SponsorID`